### PR TITLE
Use Property<LogicalLength> instead of Property<Coord>

### PR DESCRIPTION
**Pull request by tronical**
This removes the special code for the generated property getters and ensures type safety in the run-time library for property value setting.

In the Rust generated code we continue to do arithmetic on the scalar values, that means we immediately extract the scalar, do arithmetic and rely on the compiler to only allow compatible units.

Danger zone alert: In the interpreter Value::Number can now be converted to LogicalLength as-is.

### CODE REVIEW

**Comment by tronical** [comment]
This doesn't really simplify the code so much any more but rather makes it maybe a little more complicated/verbose. That said, it's certainly the clean solution to declare `Property<LogicalLength>` throughout.

**Comment by tronical** [comment]
I also played with changing LayoutInfo to contain LogicalLength but that's becoming rather messy as well without much evident benefit.

**Comment by tronical** [comment]
Rebased due to conflicts.

File: internal/backends/qt/qt_widgets/lineedit.rs [hunk 1]
```diff
@@ -80,8 +83,12 @@ impl Item for NativeLineEdit {
...
             min: match orientation {
-                Orientation::Horizontal => self.native_padding_left() + self.native_padding_right(),
-                Orientation::Vertical => self.native_padding_top() + self.native_padding_bottom(),
+                Orientation::Horizontal => {
+                    (self.native_padding_left() + self.native_padding_right()).get()
+                }
+                Orientation::Vertical => {
+                    (self.native_padding_top() + self.native_padding_bottom()).get()
+                }
             },
```
Comment (ogoffart) [inline internal/backends/qt/qt_widgets/lineedit.rs:92]: (style) I wonder if this works?

```suggestion
                Orientation::Horizontal => self.native_padding_left() + self.native_padding_right(),
                Orientation::Vertical => self.native_padding_top() + self.native_padding_bottom(),
            }.get(),
```

(just to save a few lines :-p)

**Review by ogoffart — APPROVED** [review]

File: internal/backends/qt/qt_widgets/lineedit.rs (hunk 1 above)
Comment (tronical) [reply internal/backends/qt/qt_widgets/lineedit.rs:92]: Good idea. I'll apply that throughout.

**Pull request merged**